document/core/appendix/properties.rst

```diff
@@ -670,25 +670,24 @@ To that end, all previous typing judgements :math:`C \vdash \X{prop}` are genera
 
 * The :ref:`values <syntax-val>` :math:`\val^\ast` must be of type :math:`[t_0^\ast]`.
 
-* The label type :math:`C.\CLABELS[0]` must be defined in the context.
-
-* Let :math:`C'` be the same :ref:`context <context>` as :math:`C`, but with the :ref:`label type <syntax-labeltype>` :math:`C.\CLABELS[0]` modified to :math:`(\LCATCH~[t^\ast])` in the |CLABELS| vector.
+* Let :math:`C'` be the same :ref:`context <context>` as :math:`C`, but with the label :math:`(\LCATCH~[t^\ast])` prepended to the |CLABELS| vector.
```

Comment:
C is an (implicit) _input_ to the rules, so you'll have to write the prose the other way round, i.e., first check that the last entry is `[t*]`, then define C' to be C with that entry updated. (That means that the C in the prose and the C in the formal rule denote different contexts, but I think that's fine.)

Reply:
```suggestion
* Let :math:`C'` be the same :ref:`context <context>` as :math:`C`, but with the label type :math:`(\LCATCH~[t^\ast])` prepended to the |CLABELS| vector.
```
Isn't `type` missing? All other similar clauses have it.

@rossberg Not sure if I understand. Also I wonder if making `C`'s meaning different from that in the equation will be confusing.. Can we change the equation to use the same symbol?

Reply:
Yes, we could use C' in the formal rule, though the tick would just be noise there. In general, I have opted to not artificially complicate the formal rules for the sake of the prose, though this case is minor enough.

Adding `type` sounds okay be me.

Reply:
Then can we change the symbol in the prose?

Reply:
Actually I'm not sure if I understood your comment correctly:
> C is an (implicit) _input_ to the rules, so you'll have to write the prose the other way round, i.e., first check that the last entry is `[t*]`, then define C' to be C with that entry updated. 

Can you elaborate a little more on what this means? And if the symbols don't cause much confusion, that might be fine.

Reply:
In the formal rules, the context is always explicit. Usually, we just call it C, or something else, and we can even "pattern match" it in the rules, which is what happens in the rule for caught. This implicitly imposes a constraint, because the rule is only applicable if the context has that specific form.

In the prose, the context is treated as a parameter to the "algorithm". And we keep it implicit to avoid more excessive verbosity. That implicit context is available under the fixed name C. This convention is described [here](https://webassembly.github.io/spec/core/valid/conventions.html#prose-notation). A constraint on that context (e.g., the presence of a certain entry) then has to be checked explicitly.

Reply:
Moved these changes  to PR #244, reverting these changes to whatever the document had previously.

Reply:
Please ignore the last two commits (the second reverting the first) - the changes to the typing of CAUGHTadm were already reverted. 

Changes to the typing of CAUGHTadm are now done in PR #244.